core/src/jmh/java/com/uber/m3/tally/ScopeImplBenchmark.java

```diff
@@ -67,28 +62,56 @@
 
     @Benchmark
     public void scopeReportingBenchmark(BenchmarkState state) {
-        state.scope.reportLoopIteration();
+        state.reportingBenchmarkScope.reportLoopIteration();
+    }
+
+    @Benchmark
+    public void recordingWithSanitizingDashBenchmark(BenchmarkState state) {
+        state.recordTestMetrics(state.sanitizingBenchmarkScope);
+    }
+
+    @Benchmark
+    public void recordingBenchmark(BenchmarkState state) {
+        state.recordTestMetrics(state.recordingBenchmarkScope);
     }
 
     @State(org.openjdk.jmh.annotations.Scope.Benchmark)
     public static class BenchmarkState {
 
-        private ScopeImpl scope;
+        private ScopeImpl reportingBenchmarkScope;
+        private ScopeImpl sanitizingBenchmarkScope;
+        private ScopeImpl recordingBenchmarkScope;
 
         @Setup
         public void setup() {
-            this.scope =
-                    (ScopeImpl) new RootScopeBuilder()
-                            .reporter(new TestStatsReporter())
-                            .tags(
-                                    ImmutableMap.of(
-                                            "service", "some-service",
-                                            "application", "some-application",
-                                            "instance", "some-instance"
-                                    )
-                            )
-                            .reportEvery(Duration.MAX_VALUE);
+            final ScopeBuilder scopeBuilder = new RootScopeBuilder()
+                .reporter(new TestStatsReporter())
+                .tags(
+                    ImmutableMap.of(
+                        "service", "some-service",
+                        "application", "some-application",
+                        "instance", "some-instance"
+                    )
+                );
+
+            this.reportingBenchmarkScope =
+                (ScopeImpl) scopeBuilder.reportEvery(Duration.MAX_VALUE);
+
+            this.recordTestMetrics(this.reportingBenchmarkScope);
+
+            this.recordingBenchmarkScope = (ScopeImpl) scopeBuilder.reportEvery(Duration.MAX_VALUE);
+            this.sanitizingBenchmarkScope =
+                (ScopeImpl) scopeBuilder.sanitizer(
+                    new ScopeSanitizerBuilder()
+                        .withNameValidCharacters(ValidCharacters.of(null, ValidCharacters.UNDERSCORE_CHARACTERS))
+                        .withTagKeyValidCharacters(ValidCharacters.of(null, ValidCharacters.UNDERSCORE_CHARACTERS))
+                        .withTagValueValidCharacters(ValidCharacters.of(null, ValidCharacters.UNDERSCORE_CHARACTERS))
+                        .build()
+                )
+                    .reportEvery(Duration.MAX_VALUE);
+        }
 
+        public void recordTestMetrics(final ScopeImpl scope) {
```

Comment:
I don't think this is a good routine to check -- this was used as pre-init seq, and i would suggest to keep it as such.

Instead create separate benchmarks and routines to update counter/gauge/histogram separately.

Reply:
Same as my pervious response. I appreciate your time reviewing and commenting it. 
But it would be nice if you can help the benchmarking.

Reply:
Please check my comment above. We're more than happy to help w/ guidance, but benchmarking is now a standard requirement from non-trivial contributions.

Reply:
Sure, will revert that one back. 

Can you give me details about what you want to benchmark? -- which method/function/routines?